### PR TITLE
Add vscode info

### PR DESCRIPTION
**Pull request by dveeden**
setup-an-ide: Add VS Code

### CODE REVIEW

**Comment by dveeden** [comment]
cc @dragonly

File: src/get-started/setup-an-ide.md [hunk 1]
```diff
@@ -107,3 +109,30 @@ The third config is `unit test`, which enables you can run/debug unit tests. You
...
+![VS Code with TiDE](../img/vscode_tide.png)
+
+### Prerequisites
+
+* `go`: TiDB is a Go project thus its building requires a working `go` environment. See the previous [Install Golang](install-golang.md) section to prepare the environment.
+* TiDB source code: See the previous [Get the code, build and run](build-tidb-from-source.md) section to get the source code.
+
+### Download VS Code
+
+Download GoLand from [here](https://code.visualstudio.com/Download) and install it.
```
Comment (xxchan) [inline src/get-started/setup-an-ide.md:126]:
```suggestion
Download VS Code from [here](https://code.visualstudio.com/Download) and install it.
```

File: src/get-started/setup-an-ide.md [hunk 2]
```diff
@@ -107,3 +109,30 @@ The third config is `unit test`, which enables you can run/debug unit tests. You
...
+
+Now install these extensions:
+
+* [Go](https://marketplace.visualstudio.com/items?itemName=golang.Go)
+* [TiDE](https://marketplace.visualstudio.com/items?itemName=dragonly.ticode)
+* [GitHub Pull Requests and Issues](https://marketplace.visualstudio.com/items?itemName=github.vscode-pull-request-github)
```
Comment (xxchan) [inline src/get-started/setup-an-ide.md:132]:
```suggestion
```
I think this extension is irrelevant?

File: src/get-started/setup-an-ide.md [hunk 3]
```diff
@@ -107,3 +109,30 @@ The third config is `unit test`, which enables you can run/debug unit tests. You
...
+* TiDB source code: See the previous [Get the code, build and run](build-tidb-from-source.md) section to get the source code.
+
+### Download VS Code
+
+Download GoLand from [here](https://code.visualstudio.com/Download) and install it.
+
+Now install these extensions:
+
+* [Go](https://marketplace.visualstudio.com/items?itemName=golang.Go)
+* [TiDE](https://marketplace.visualstudio.com/items?itemName=dragonly.ticode)
```
Comment (xxchan) [inline src/get-started/setup-an-ide.md:131]: And I think TiDE is also not mandatory? 

Also, GoLand instructions above mainly focus on simple tasks like unistore and unit tests. But TiDE is targeted at more complex tasks (and doesn't support unistore)?

File: src/get-started/setup-an-ide.md (hunk 2 above)
Comment (dveeden) [reply src/get-started/setup-an-ide.md:132]: I think having this extension makes sense as GitHub is core the the contribution workflow. We could mark it as optional, but then we should probably label TiDE as optional as well.

File: src/get-started/setup-an-ide.md (hunk 3 above)
Comment (dveeden) [reply src/get-started/setup-an-ide.md:131]: Yes not mandatory, but quite useful. I think users would be smart enough to filter this out if needed and if they don't that's also not really a problem.

File: src/get-started/setup-an-ide.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # Setup an IDE
 
+Using an IDE is recommended as it makes it a lot easier to for example see the fields of a struct. However it is not required to use a specic IDE or editor.
```
Comment (tisonkun) [inline src/get-started/setup-an-ide.md:3]: what is "it is not required to use a specic IDE or editor"? even change "specic" to "specific" it still confuses me.

Comment (dveeden) [reply src/get-started/setup-an-ide.md:3]:
```suggestion
Using an IDE is recommended as it makes it a lot easier to work with the TiDB code, for example to see the fields of a struct. However it is not required to use a specific IDE or editor. You can use the IDE or editor of your choice.
```

Comment (tisonkun) [reply src/get-started/setup-an-ide.md]: Thanks for your update. I get the point.

**Comment by tisonkun** [comment]
Merging...

Thanks for your contribution @dveeden !

**Pull request merged**